Public/Src/Cache/ContentStore/Distributed/NuCache/RedisGlobalStore.cs

```diff
@@ -703,11 +703,11 @@ public Task<BoolResult> InvalidateLocalMachineAsync(OperationContext context)
         }
 
         /// <inheritdoc />
-        public Task<BoolResult> PutBlobAsync(OperationContext context, ContentHash hash, byte[] blob)

```

Comment:
Any particular reason to make this method async?

Reply:
yep: type conversion. This method calls PutBlobAsync that returns a type that derives from BoolResult.
There is no covarient conversion from 'Task<DerivedType>' to 'Task<BaseType>'. In order to get it you have to await the result.